x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/accounts_stats_collector.ts

```diff
@@ -0,0 +1,181 @@
+// TODO: naming, naming, naming
+
+export interface Accounts {
+  accounts: {
+    buckets: AccountEntity[];
+  };
+}
+
+export interface AccountEntity {
+  key: string; // account_id
+  doc_count: number;
+  resource_type: {
+    buckets: ResourceType[];
+  };
+}
+
+export interface ResourceType {
+  key: string;
+  doc_count: number;
+  evaluation: { buckets: StatsEntity[] };
+}
+
+export interface StatsEntity {
+  key: string;
+  doc_count: number;
+  agents: Agents;
+}
+export interface Agents {
+  value: number;
+}
+
+export interface ResourceStats {
+  [key: string]: {
+    doc_count: number;
+    passed: number;
+    failed: number;
+  };
+}
+
+export const getAccountsStats = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<CspmAccountsStats[]> => {
+  try {
+    const isIndexExists = await esClient.indices.exists({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+    });
+
+    if (isIndexExists) {
+      return await getAccountStatsByResourceType(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS);
+    }
+
+    return [];
+  } catch (e) {
+    logger.error(`Failed to get accounts stats ${e}`);
+    return [];
+  }
+};
+
+const getClusterScore = (evaluationByResourceType: ResourceStats): number => {
+  let passed = 0;
+  let failed = 0;
+  Object.entries(evaluationByResourceType).forEach(([key, value], index) => {
+    passed += value.passed ? value.passed : 0;
+    failed += value.failed ? value.failed : 0;
+  });
+  return calculatePostureScore(passed, failed);
+};
+
+const getAccountStatsByResourceType = async (
+  esClient: ElasticsearchClient,
+  index: string
+): Promise<CspmAccountsStats[]> => {
+  let agents = 1;
+  const accountsStats = await esClient.search<unknown, Accounts>(getAccountStatsQuery(index));
+
+  const accountsBucket = accountsStats.aggregations?.accounts.buckets
+    ? accountsStats.aggregations?.accounts.buckets
+    : [];
+
+  const accounts = accountsBucket.map((account) => {
+    const resourcesBucket = account.resource_type?.buckets ? account.resource_type?.buckets : [];
+
+    const evaluationByResourceType: ResourceStats[] = resourcesBucket.map(
+      (resourceType: ResourceType) => {
+        const evaluationBucket = resourceType.evaluation.buckets
+          ? resourceType.evaluation.buckets
+          : [];
+
+        const byEvaluation = evaluationBucket.map((evaluation: StatsEntity) => {
+          agents = Math.max(agents, evaluation.agents.value);
+          return { [evaluation.key]: evaluation.doc_count };
+        });
+
+        return {
+          [resourceType.key]: {
+            doc_count: resourceType.doc_count,
+            ...Object.assign({}, ...byEvaluation),
+          },
+        };
+      }
+    );
+
+    const parsedEvaluationByResourceType = evaluationByResourceType
+      ? Object.fromEntries(evaluationByResourceType.map((x) => Object.entries(x)[0]))
+      : {};
+    const accountScore = getClusterScore(parsedEvaluationByResourceType);
```

Comment:
looks like we'll pass `{}` to `getClusterScore` if we don't have `evaluationByResourceType`, and passing it will result in `calculatePostureScore(0,0)`, which i don't think we intend to do?

in anyway, also looks like it will always be truthy as it's an array. so not sure what is intended here. could you clarify ?